app/models/order_balance.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class OrderBalance
+  def initialize(order)
```

Comment:
So with this in place, we can make changes like this:
```diff
-order.outstanding_balance
+OrderBalance.new(order).to_f
```

Right? And in a lot of places we can omit the `to_f` because that's called anyway. It may be worth it delegating `to_s` to `to_f` as well.

Reply:
If we didn't have a dependency to `current_user` we could even replace `Spree::Order#outstanding_balance` to use this new class and we had a drop-in replacement without needing to update any other code.

Reply:
> And in a lot of places we can omit the to_f because that's called anyway. It may be worth it delegating to_s to to_f as well.

Yeah, I was actually wondering if we could leverage that implicit conversion. It'd be ace but it didn't succeed when I tried.

> If we didn't have a dependency to current_user we could even replace Spree::Order#outstanding_balance to use this new class and we had a drop-in replacement without needing to update any other code.

That's even more interesting because I wouldn't need to change any callers, yes! I'll see if I manage to do so.

Reply:
I gave it a try and we might be lucky. See my last two commits in https://github.com/openfoodfoundation/openfoodnetwork/pull/6959. Indeed, adding a `to_s` made the trick when rendering `OrderBalance` objects in the view.